modules/core/lib/utils/network_manager/network_change_manager.dart

```diff
@@ -0,0 +1,61 @@
+import 'dart:async';
+
+import 'package:connectivity_plus/connectivity_plus.dart';
+
+typedef NetworkCallBack = void Function(NetworkResult result);
+
+abstract class INetworkChangeManager {
```

Comment:
To comply with the project's interface structure, let's remove the 'I' at the beginning and open it as a separate file.